2.4/root/usr/bin/run-mongod

```diff
@@ -46,8 +32,8 @@ function cleanup() {
     mongo_remove
   fi
   echo "=> Shutting down MongoDB server ..."
-  if [ -f "${MONGODB_PID_FILE}" ]; then
-    kill -2 $(cat ${MONGODB_PID_FILE})
+  if [[ -s ${MONGODB_DATADIR}/mongod.lock ]]; then
+    kill -2 $(cat ${MONGODB_DATADIR}/mongod.lock)
```

Comment:
> Do you know why cleanup() uses kill or pkill instead of mongod --shutdown?

I don't know, probably using `--shutdown` would be preferred. But again, it's easier to review PRs that do one thing.
That change would be much easier to review in a PR of itself IMHO.